### PR TITLE
refactor whole plugin

### PR DESCRIPTION
**Pull request by JulianMar**
easier use now.
Module stuff was deleted
less Settings
make path resolving sync
remove unneeded dependencies

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -39,62 +24,33 @@ export class ThemeResolverPlugin {
...
-            "ThemeResolverPlugin",
-            (request: any, resolveContext: any, callback: () => void) => {
-                this.pathRegex.forEach((reg, x) => {
-                    if (request.request.match(reg)) {
-                        this.chosenResolver = Object.assign(ThemeResolverPlugin.defaultOptions, this.options[x]);
+        const target = resolver.ensureHook("resolved");
+
+        resolver.getHook("module")
+        .tapAsync("ThemeResolverPlugin", (request: any, resolveContext: any, callback: () => void) => {
+                let chosenResolver = this.getResolver(request);
```
Comment (mwidmann) [inline src/index.ts]: shoudn't this be a `const`? It's not replaced anywhere afterwards.

File: src/index.ts [hunk 2]
```diff
@@ -39,62 +24,33 @@ export class ThemeResolverPlugin {
...
-        resolver.hooks.module.tapAsync(
-            "ThemeResolverPlugin",
-            (request: any, resolveContext: any, callback: () => void) => {
-                this.pathRegex.forEach((reg, x) => {
-                    if (request.request.match(reg)) {
-                        this.chosenResolver = Object.assign(ThemeResolverPlugin.defaultOptions, this.options[x]);
+        const target = resolver.ensureHook("resolved");
+
+        resolver.getHook("module")
+        .tapAsync("ThemeResolverPlugin", (request: any, resolveContext: any, callback: () => void) => {
```
Comment (mwidmann) [inline src/index.ts]: Can we somehow figure out what the types for `request` and `resolveContext` are?

File: src/index.ts [hunk 3]
```diff
@@ -103,60 +59,34 @@ export class ThemeResolverPlugin {
...
-                            const res = moduleresolver.sync(dep, {basedir: process.cwd()});
-                            if (res) {
-                                resolve(res);
-                            }
-                        } catch (e) {
-                        reject(new Error("Module is not resolvable"));
-                        }
-                    },
-                );
+    private getResolver(request: any): IThemeResolverPluginOptions | void
```
Comment (mwidmann) [inline src/index.ts]: Same here. If we figure out the type of `request`, it should apply here as well.

File: src/index.ts (hunk 2 above)
Comment (mwidmann) [reply src/index.ts]: This should help: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/enhanced-resolve/index.d.ts

File: src/index.ts (hunk 1 above)
Comment (JulianMar) [reply src/index.ts]: yes this should be a const

File: src/index.ts (hunk 2 above)
Comment (JulianMar) [reply src/index.ts]: Sadly the types are outdated. They are generated for version 3.0 and currently there is a version 5 in beta.

**Comment by JulianMar** [comment]
I updated the PR with some hints by tslint and your recommendation to change the let to a const

**Pull request merged**